task_processing/plugins/mesos/logging_executor.py

```diff
@@ -0,0 +1,151 @@
+import json
+import logging
+import sys
+import time
+import urllib.request
+from threading import Thread
+
+from pyrsistent import field
+from pyrsistent import m
+from pyrsistent import PMap
+from pyrsistent import pmap
+from pyrsistent import PRecord
+from six.moves.queue import Queue
+
+from task_processing.interfaces.task_executor import TaskExecutor
+
+log = logging.getLogger(__name__)
+
+# Read task log in 4K chunks
+TASK_LOG_CHUNK_LEN = 4096
+
+
+class LogMetadata(PRecord):
+    log_url = field(type=str, initial='')
+    log_offsets = field(type=PMap,
+                        factory=pmap,
+                        initial=pmap({'stdout': 0, 'stderr': 0}))
+
+
+class LoggingExecutor(TaskExecutor):
+    def __init__(self, downstream_executor):
+        self.downstream_executor = downstream_executor
+
+        self.src_queue = downstream_executor.get_event_queue()
+        self.dest_queue = Queue()
+        self.stopping = False
+
+        # No lock is needed because log_metadata is touched in a single thread
+        self.log_metadata = m()
+
+        self.logging_thread = Thread(target=self.logging_loop)
+        self.logging_thread.daemon = True
+        self.logging_thread.start()
+
+    def build_task_log_url(self, task_log_url, event):
+        try:
+            response = json.load(urllib.request.urlopen(
```

Comment:
minor nit: can we use requests instead of urllib?

Reply:
Will fix.